streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -255,8 +255,9 @@ public ByteBuffer subscriptionUserData(final Set<String> topics) {
             taskManager.processId(),
             userEndPoint,
             taskManager.getTaskOffsetSums(),
-            uniqueField)
-                .encode();
+            uniqueField,
+            (byte) assignmentErrorCode.get()
```

Comment:
I think I'd like to re-raise Sophie's concern here. It doesn't compute for me why we are casting an int to a byte here..

Reply:
I guess I must have misunderstood  your earlier comment. I thought you wanted it to stay a byte so that is why I pushed back. But if you have no objections I will just change it